983/remove assetsconfig from other pallets

HaidarJbeily7 opened the pull request:
- Replaced usage of assets pallet public functions with AssetManager trait
- Removed assets::Config from the following pallets:
  - ceres-launchpad
  - ceres-token-locker
  - ceres-liquidity-locker
  - ceres-governance-platform
  - demeter-farming-platform
  - pool-xyk
  - price-tools 
  - technical
  - farming
  - vested-rewards
  - rewards
  - faucet
  - kensetsu
  - bridge-proxy
  - qa-tools
  - order-book
  - xst

wer1st reviewed (commented):
You have to remove `assets` from all `Cargo.toml` files of pallets.
If you need `assets` dependency for tests / mock, please move it to `[dev-dependencies]`

HaidarJbeily7 commented:
I double checked all cargo.toml files and fixed where the issue of not removing assets dependency appears..

Also, I removed the assets config from the rest pallets:

- ceres-staking
- apollo-platform
- dex-api
- dex-manager
- hemes-governance-platform
- iroha-migration
- xor-fee
- referrals
- trading-pair
- pswap-distribution


FYI @wer1st

wer1st commented on pallets/dex-manager/Cargo.toml:
```diff
@@ -46,6 +46,7 @@ std = [
     'codec/std',
     "scale-info/std",
     'serde/std',
+    'assets/std',
```
delete

wer1st commented on pallets/farming/Cargo.toml:
```diff
@@ -29,7 +29,7 @@ sp-core = { git = "https://github.com/sora-xor/substrate.git", branch = "polkado
 sp-std = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 
-assets = { path = "../assets", default-features = false }
+
```
delete empty line

wer1st commented on pallets/pool-xyk/Cargo.toml:
```diff
@@ -30,7 +30,7 @@ sp-runtime = { git = "https://github.com/sora-xor/substrate.git", branch = "polk
 pallet-timestamp = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 
-assets = { path = "../assets", default-features = false }
+
```
delete

wer1st commented on pallets/pool-xyk/src/action_deposit_liquidity.rs:
```diff
@@ -33,11 +33,11 @@ use frame_support::weights::Weight;
 use crate::{to_balance, to_fixed_wrapper, AccountPools, PoolProviders, TotalIssuances};
 
-use crate::aliases::{AccountIdOf, AssetIdOf, TechAccountIdOf};
+use common::prelude::{AssetIdOf, AssetInfoProvider, Balance, FixedWrapper};
```
Move it to line 35

wer1st commented on pallets/pool-xyk/src/action_poly_swap.rs:
```diff
@@ -32,11 +32,11 @@ use frame_support::dispatch;
+use crate::aliases::{AccountIdOf, DEXIdOf, PolySwapActionStructOf, TechAccountIdOf};
 use crate::Config;
 
+use common::{AssetIdOf, SwapRulesValidation};
```
Move it to line 34

wer1st commented on pallets/technical/Cargo.toml:
```diff
@@ -29,7 +29,7 @@ sp-runtime = { git = "https://github.com/sora-xor/substrate.git", branch = "polk
 twox-hash = { version = "1.5.0", default-features = false }
 
-assets = { path = "../assets", default-features = false }
+
```
delete

wer1st commented on pallets/vested-rewards/Cargo.toml:
```diff
@@ -11,7 +11,7 @@ version = '1.2.1'
 
 [dependencies]
-assets = { path = "../assets", default-features = false }
+
```
delete

wer1st commented on pallets/xor-fee/Cargo.toml:
```diff
@@ -31,7 +31,7 @@ sp-std = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot
 traits = { git = "https://github.com/open-web3-stack/open-runtime-module-library.git", package = "orml-traits", default-features = false }
 smallvec = "1.5.0"
-assets = { path = "../assets", default-features = false }
+
```
delete

wer1st commented on common/src/traits.rs:
```diff
@@ -1089,6 +1143,232 @@ impl<AssetId, AccountId, AssetSymbol, AssetName, BalancePrecision, ContentSource
+    ) -> DispatchResult;
+
+    fn burn_from(
+        asset_id: Self::AssetId,
```
```suggestion
        asset_id: &Self::AssetId,
```

wer1st commented on common/src/traits.rs:
```diff
@@ -1089,6 +1143,232 @@ impl<AssetId, AccountId, AssetSymbol, AssetName, BalancePrecision, ContentSource
+    ) -> DispatchResult;
+
+    fn mint_to(
+        asset_id: Self::AssetId,
```
```suggestion
        asset_id: &Self::AssetId,
```
